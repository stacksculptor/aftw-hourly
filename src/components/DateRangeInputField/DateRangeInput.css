@import '../../marketplace.css';

:root {
  --selectionHeight: 36px;
  --hoveredOverlayColor: rgba(255, 255, 255, 0.2);
  --datepickerDistanceToTop: 252px;
}

.inputRoot {

  /* Contain repainting to this component only */
  transform: translate3d(0,0,0);

  /* Override react-dates default styles to match input styles */

  & :global(.DateRangePicker) {
    display: block;
  }
  & :global(.DateRangePicker__picker--horizontal) {
    top: 36px;
    width: 100%;
    min-height: calc(100vh - var(--datepickerDistanceToTop));
    background-color: var(--marketplaceColor);
  }
  & :global(.DateRangePickerInput) {
    width: 100%;
    display: flex;
    justify-content: space-between;
    border: none;
  }
  & :global(.DayPicker--horizontal) {
    margin: 0 auto;
    background-color: var(--marketplaceColor);
    box-shadow: none;
  }
  & :global(.CalendarMonthGrid) {
    background-color: transparent;
  }
  & :global(.DateInput) {
    display: block;
    width: 100%;
    padding: 5px 0 2px 0;
  }
  & :global(.DayPicker__week-header) {
    color: var(--matterColorLight);
    top: 57px;
  }

  & :global(.DayPicker__week-header li) {
    font-weight: 400;
  }

  & :global(.CalendarMonth__caption) {
    color: var(--matterColorLight);
    @apply --marketplaceH2FontStyles;
    margin: 1px 0 14px;
    font-weight: 400;
  }
  & :global(.DayPickerNavigation__prev),
  & :global(.DayPickerNavigation__next) {
    color: var(--matterColorLight);
  }
  & :global(.CalendarDay) {
    @apply --marketplaceH4FontStyles;
    color: var(--matterColorLight);
    border: 0;
    margin-top: 0;
    margin-bottom: 0;

    @media (--desktopViewport) {
      margin-top: 0;
      margin-bottom: 0;
    }
  }

  /* Add an underline for '.renderedDay' */
  & :global(.CalendarDay--today) {
    background-image: url("data:image/svg+xml;utf8,<svg width='14' height='2' viewBox='0 0 14 2' xmlns='http://www.w3.org/2000/svg'><path d='M0 0h14v2H0z' fill='#FFF' fill-rule='evenodd'/></svg>");
    background-position: center 30px;
  }

  /* Remove default bg-color and use our extra span instead '.renderedDay' */
  & :global(.CalendarDay--hovered-span),
  & :global(.CalendarDay--selected-span) {
    background-image: transparent;
    background-color: transparent;
  }
  & :global(.CalendarDay--hovered-span .renderedDay) {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: var(--selectionHeight);
    background-color: var(--hoveredOverlayColor);
  }
  & :global(.CalendarDay--selected-span .renderedDay) {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: var(--selectionHeight);
    background-color: var(--successColor);
    transition: all 0.2s ease-out;
  }
  /* Remove default bg-color and use our extra span instead '.renderedDay' */
  & :global(.CalendarDay--selected-start) {
    background-color: transparent;
    background-image: none;
  }
  & :global(.CalendarDay--selected-start .renderedDay) {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: var(--selectionHeight);
    background-color: var(--successColor);
    border-top-left-radius: calc(var(--selectionHeight) / 2);
    border-bottom-left-radius: calc(var(--selectionHeight) / 2);
  }
  /* Remove default bg-color and use our extra span instead '.renderedDay' */
  & :global(.CalendarDay--after-hovered-start) {
    background-color: transparent;
  }
  & :global(.CalendarDay--after-hovered-start .renderedDay) {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: var(--selectionHeight);
    background-color: var(--hoveredOverlayColor);
  }
  /* Remove default bg-color and use our extra span instead '.renderedDay' */
  & :global(.CalendarDay--selected-end) {
    background-color: transparent;
  }
  & :global(.CalendarDay--selected-end .renderedDay) {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: var(--selectionHeight);
    background-color: var(--successColor);
    border-top-right-radius: calc(var(--selectionHeight) / 2);
    border-bottom-right-radius: calc(var(--selectionHeight) / 2);
  }
  /* Remove default bg-color and use our extra span instead '.renderedDay' */
  & :global(.CalendarDay--hovered) {
    background-color: transparent;
  }
  & :global(.CalendarDay--hovered .renderedDay) {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: var(--selectionHeight);
    background-color: var(--hoveredOverlayColor);
  }
  /* Remove default bg-color and use our extra span instead '.renderedDay' */
  & :global(.CalendarDay--blocked-out-of-range) {
    background-color: transparent;
    color: var(--marketplaceColorDark);
    border: 0;
  }
  & :global(.CalendarDay--blocked-out-of-range .renderedDay) {
    background-color: transparent;
  }
  & :global(.DateInput--with-caret::before) {
    display: none;
  }
  & :global(.DateInput--with-caret::after) {
    display: none;
  }
  & :global(.DateInput__display-text) {
    @apply --marketplaceDefaultFontStyles;
    padding: 0;
    color: var(--matterColorAnti);
  }
  & :global(.DateInput__display-text--focused) {
    background-color: transparent;
  }
  & :global(.DateInput__display-text--has-input) {
    color: var(--matterColor);
  }
}

/**
 * Mobile margins mean that labels, inputs (incl separate borders) get margins,
 * but calendar popup doesn't.
 */
.withMobileMargins {
  & :global(.DateRangePickerInput) {
    width: calc(100% - 48px);
    margin: 0 24px 3px 24px;
  }

  /* Create gutter between inputs */
  & :global(.DateInput) {
    width: calc(100% - 6px);
  }
  & :global(.DateInput:first-of-type) {
    margin-right: 12px;
  }
}

.rootNextMonthIcon,
.rootPreviousMonthIcon {
  stroke: var(--matterColorLight);
  fill: var(--matterColorLight);
}
