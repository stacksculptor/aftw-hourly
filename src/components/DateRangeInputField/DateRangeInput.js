/**
 * DateRangeInput wraps DateRangePicker from React-dates and gives a list of all default props we use.
 * Styles for DateRangePicker can be found from 'public/reactDates.css'.
 *
 * N.B. *isOutsideRange* in defaultProps is defining what dates are available to booking.
 */
import React, { Component, PropTypes } from 'react';
import { intlShape, injectIntl } from 'react-intl';
import { DateRangePicker, isInclusivelyAfterDay, isInclusivelyBeforeDay } from 'react-dates';
import classNames from 'classnames';
import moment from 'moment';

import NextMonthIcon from './NextMonthIcon';
import PreviousMonthIcon from './PreviousMonthIcon';
import css from './DateRangeInput.css';

export const START_DATE = 'startDate';
export const END_DATE = 'endDate';
export const HORIZONTAL_ORIENTATION = 'horizontal';
export const ANCHOR_LEFT = 'left';

// Since redux-form tracks the onBlur event for marking the field as
// touched (which triggers possible error validation rendering), only
// trigger the event asynchronously when no other input within this
// component has received focus.
//
// This prevents showing the validation error when the user selects a
// value and moves on to another input within this component.
const BLUR_TIMEOUT = 100;

// Possible configuration options of React-dates
const defaultProps = {
  initialDates: null, // Possible initial date passed for the component
  value: null, // Value should keep track of selected date.

  // input related props
  startDateId: 'startDate',
  endDateId: 'endDate',
  startDatePlaceholderText: null, // Handled inside component
  endDatePlaceholderText: null, // Handled inside component
  disabled: false,
  required: false,
  screenReaderInputMessage: null, // Handled inside component
  showClearDates: false,
  showDefaultInputIcon: false,
  customArrowIcon: <span />,
  customInputIcon: null,
  customCloseIcon: null,

  // calendar presentation and interaction related props
  orientation: HORIZONTAL_ORIENTATION,
  anchorDirection: ANCHOR_LEFT,
  horizontalMargin: 0,
  withPortal: false,
  withFullScreenPortal: false,
  initialVisibleMonth: null,
  numberOfMonths: 1,
  keepOpenOnDateSelect: false,
  reopenPickerOnClearDates: false,
  renderCalendarInfo: null,
  daySize: 38,

  // navigation related props
  navPrev: <PreviousMonthIcon />,
  navNext: <NextMonthIcon />,
  onPrevMonthClick() {},
  onNextMonthClick() {},

  // day presentation and interaction related props
  renderDay: day => {
    return <span className="renderedDay">{day.format('D')}</span>;
  },
  minimumNights: 1,
  enableOutsideDays: false,
  isDayBlocked: () => false,

  // Stripe holds funds in a reserve for up to 90 days from charge creation.
  // outside range -><- today ... today+89 days -><- outside range
  isOutsideRange: day => {
    const daysCountAvailableToBook = 90;
    const endOfRange = daysCountAvailableToBook - 1;
    return !isInclusivelyAfterDay(day, moment()) ||
      !isInclusivelyBeforeDay(day, moment().add(endOfRange, 'days'));
  },
  isDayHighlighted: () => {},

  // Internationalization props
  // Multilocale support can be achieved with displayFormat like moment.localeData.longDateFormat('L')
  // https://momentjs.com/
  displayFormat: 'ddd, MMMM D',
  monthFormat: 'MMMM YYYY',
  phrases: {
    closeDatePicker: null, // Handled inside component
    clearDate: null, // Handled inside component
  },
};

class DateRangeInputComponent extends Component {
  constructor(props) {
    super(props);

    this.state = {
      focusedInput: null,
    };

    this.blurTimeoutId = null;
    this.onDatesChange = this.onDatesChange.bind(this);
    this.onFocusChange = this.onFocusChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    // Update focusedInput in case a new value for it is
    // passed in the props. This may occur if the focus
    // is manually set to the date picker.
    if (nextProps.focusedInput && nextProps.focusedInput !== this.props.focusedInput) {
      this.setState({ focusedInput: nextProps.focusedInput });
    }
  }

  componentWillUnmount() {
    window.clearTimeout(this.blurTimeoutId);
  }

  onDatesChange(dates) {
    const { startDate, endDate } = dates;
    const startDateAsDate = startDate instanceof moment ? startDate.toDate() : null;
    const endDateAsDate = endDate instanceof moment ? endDate.toDate() : null;
    this.props.onChange({ startDate: startDateAsDate, endDate: endDateAsDate });
  }

  onFocusChange(focusedInput) {
    // DateRangePicker requires 'onFocusChange' function and 'focusedInput'
    // but Fields of React-Form deals with onFocus & onBlur instead
    this.setState({ focusedInput });

    if (focusedInput) {
      window.clearTimeout(this.blurTimeoutId);
      this.props.onFocus(focusedInput);
    } else {
      window.clearTimeout(this.blurTimeoutId);
      this.blurTimeoutId = window.setTimeout(
        () => {
          this.props.onBlur();
        },
        BLUR_TIMEOUT
      );
    }
  }

  render() {
    /* eslint-disable no-unused-vars */
    const {
      className,
      initialDates,
      intl,
      name,
      startDatePlaceholderText,
      endDatePlaceholderText,
      onBlur,
      onChange,
      onFocus,
      onDragStart,
      onDrop,
      phrases,
      screenReaderInputMessage,
      useMobileMargins,
      value,
      ...datePickerProps
    } = this.props;
    /* eslint-enable no-unused-vars */

    const initialStartMoment = initialDates ? moment(initialDates.startDate) : null;
    const initialEndMoment = initialDates ? moment(initialDates.endDate) : null;
    const startDate = value && value.startDate instanceof Date
      ? moment(value.startDate)
      : initialStartMoment;
    const endDate = value && value.endDate instanceof Date
      ? moment(value.endDate)
      : initialEndMoment;

    const startDatePlaceholderTxt = startDatePlaceholderText ||
      intl.formatMessage({ id: 'DateRangeInputField.startDatePlaceholderText' });
    const endDatePlaceholderTxt = endDatePlaceholderText ||
      intl.formatMessage({ id: 'DateRangeInputField.endDatePlaceholderText' });
    const screenReaderInputText = screenReaderInputMessage ||
      intl.formatMessage({ id: 'DateRangeInputField.screenReaderInputMessage' });
    const closeDatePickerText = phrases.closeDatePicker
      ? phrases.closeDatePicker
      : intl.formatMessage({ id: 'DateRangeInputField.closeDatePicker' });
    const clearDateText = phrases.clearDate
      ? phrases.clearDate
      : intl.formatMessage({ id: 'DateRangeInputField.clearDate' });

    const classes = classNames(css.inputRoot, className, {
      [css.withMobileMargins]: useMobileMargins,
    });

    return (
      <div className={classes}>
        <DateRangePicker
          {...datePickerProps}
          focusedInput={this.state.focusedInput}
          onFocusChange={this.onFocusChange}
          startDate={startDate}
          endDate={endDate}
          onDatesChange={this.onDatesChange}
          startDatePlaceholderText={startDatePlaceholderTxt}
          endDatePlaceholderText={endDatePlaceholderTxt}
          screenReaderInputMessage={screenReaderInputText}
          phrases={{ closeDatePicker: closeDatePickerText, clearDate: clearDateText }}
        />
      </div>
    );
  }
}

DateRangeInputComponent.defaultProps = {
  className: null,
  useMobileMargins: false,
  ...defaultProps,
};

const { bool, func, instanceOf, oneOf, shape, string } = PropTypes;

DateRangeInputComponent.propTypes = {
  className: string,
  focusedInput: oneOf([START_DATE, END_DATE]),
  initialDates: instanceOf(Date),
  intl: intlShape.isRequired,
  name: string.isRequired,
  isOutsideRange: func,
  onChange: func.isRequired,
  onBlur: func.isRequired,
  onFocus: func.isRequired,
  onDragStart: func.isRequired,
  onDrop: func.isRequired,
  phrases: shape({
    closeDatePicker: string,
    clearDate: string,
  }),
  useMobileMargins: bool,
  startDatePlaceholderText: string,
  endDatePlaceholderText: string,
  screenReaderInputMessage: string,
  value: shape({
    startDate: instanceOf(Date),
    endDate: instanceOf(Date),
  }),
};

export default injectIntl(DateRangeInputComponent);
