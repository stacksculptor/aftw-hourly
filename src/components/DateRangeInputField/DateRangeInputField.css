@import '../../marketplace.css';

.fieldRoot {

}

.labels {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.startDateLabel,
.endDateLabel {
  flex-basis: 50%;
  @apply --marketplaceH4FontStyles;
  font-weight: 600;
  color: var(--matterColor);
  margin-top: 0;
  margin-bottom: 0;

  @media (--desktopViewport) {
    margin-top: 0;
    margin-bottom: 0;
  }
}

.labelSuccess {
  color: var(--successColor);
}

.inputBorders {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.input {
  flex-basis: 50%;
  border-bottom-style: solid;
  border-bottom-width: 3px;
  border-bottom-color: var(--attentionColor);

  transition: border-bottom-color var(--transitionStyle);

  &:hover,
  &:focus {
    border-bottom-color: var(--matterColor);
    outline: none;
  }
}

.inputSuccess {
  border-bottom-color: var(--successColor);
}

.inputError {
  border-bottom-color: var(--failColor);
}

.hover {
  border-bottom-color: var(--matterColorDark);
}

/**
 * Mobile margins mean that labels, inputs (incl separate borders) get margins,
 * but calendar popup doesn't.
 */
.mobileMargins {
  width: calc(100% - 48px);
  margin: 0 24px;

  /* Gutter between inputs (when calendar is not visible) */
  & .startDateLabel,
  & .endDateLabel {
    flex-basis: calc(50% - 9px);
  }

  & .input {
    flex-basis: calc(50% - 9px);
  }
}
