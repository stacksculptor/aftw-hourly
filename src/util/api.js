import { types as sdkTypes, transit } from './sdkLoader';
import Decimal from 'decimal.js';

const apiBaseUrl = () => {
  const port = process.env.REACT_APP_DEV_API_SERVER_PORT;
  const useDevApiServer = process.env.NODE_ENV === 'development' && !!port;

  // In development, the dev API server is running in a different port
  if (useDevApiServer) {
    return `http://localhost:${port}`;
  }

  // Otherwise, use the same domain and port as the frontend
  return '/';
};

const typeHandlers = [
  {
    type: sdkTypes.BigDecimal,
    customType: Decimal,
    writer: v => new sdkTypes.BigDecimal(v.toString()),
    reader: v => new Decimal(v.value),
  },
];

const serialize = data => {
  return transit.write(data, { typeHandlers });
};

const deserialize = str => {
  return transit.read(str, { typeHandlers });
};

const post = (path, body) => {
  const url = `${apiBaseUrl()}${path}`;
  const options = {
    method: 'POST',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/transit+json',
    },
    body: serialize(body),
  };
  return window
    .fetch(url, options)
    .then(res => {
      if (res.status >= 400) {
        const e = new Error('Local API request failed');
        e.apiResponse = res;
        throw e;
      }
      return res;
    })
    .then(res => {
      const contentTypeHeader = res.headers.get('Content-Type');
      const contentType = contentTypeHeader ? contentTypeHeader.split(';')[0] : null;
      if (contentType === 'application/transit+json') {
        return res.text().then(deserialize);
      } else if (contentType === 'application/json') {
        return res.json();
      }
      return res.text();
    });
};

export const transactionLineItems = body => {
  return post('/api/transaction-line-items', body);
};

export const initiatePrivileged = body => {
  return post('/api/initiate-privileged', body);
};
