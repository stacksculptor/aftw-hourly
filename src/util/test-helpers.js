import React from 'react';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router';
import { Provider } from 'react-redux';
import configureStore from '../store';

export const TestProvider = props => {
  const store = configureStore();
  return (
    <IntlProvider locale="en">
      <BrowserRouter>
        <Provider store={store}>
          {props.children}
        </Provider>
      </BrowserRouter>
    </IntlProvider>
  );
};
