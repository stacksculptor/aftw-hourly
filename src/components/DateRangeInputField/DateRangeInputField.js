import React, { Component, PropTypes } from 'react';
import { Field } from 'redux-form';
import classNames from 'classnames';
import { ValidationError } from '../../components';

import DateRangeInput, { START_DATE, END_DATE } from './DateRangeInput';
import css from './DateRangeInputField.css';

class DateRangeInputFieldComponent extends Component {
  constructor(props) {
    super(props);
    this.state = { focusedInput: null };
    this.handleBlur = this.handleBlur.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
  }

  handleBlur(focusedInput) {
    this.setState({ focusedInput: null });
    this.props.input.onBlur(focusedInput);
  }

  handleFocus(focusedInput) {
    this.setState({ focusedInput });
    this.props.input.onFocus(focusedInput);
  }

  render() {
    const {
      rootClassName,
      className,
      startDateId,
      startDateLabel,
      endDateId,
      endDateLabel,
      input,
      meta,
      ...rest
    } = this.props;

    if (startDateLabel && !startDateId) {
      throw new Error('startDateId required when a startDateLabel is given');
    }

    if (endDateLabel && !endDateId) {
      throw new Error('endDateId required when a endDateLabel is given');
    }

    const { touched, error } = meta;
    const value = input.value;

    // If startDate is valid label changes color and bottom border changes color too
    const startDateIsValid = value && value.startDate instanceof Date;
    const startDateLabelClasses = classNames(css.startDateLabel, {
      [css.labelSuccess]: false, //startDateIsValid,
    });
    const startDateBorderClasses = classNames(css.input, {
      [css.inputSuccess]: startDateIsValid,
      [css.inputError]: touched && !startDateIsValid && typeof error === 'string',
      [css.hover]: this.state.focusedInput === START_DATE,
    });

    // If endDate is valid label changes color and bottom border changes color too
    const endDateIsValid = value && value.endDate instanceof Date;
    const endDateLabelClasses = classNames(css.endDateLabel, {
      [css.labelSuccess]: false, //endDateIsValid,
    });
    const endDateBorderClasses = classNames(css.input, {
      [css.inputSuccess]: endDateIsValid,
      [css.inputError]: touched && !endDateIsValid && typeof error === 'string',
      [css.hover]: this.state.focusedInput === END_DATE,
    });

    const label = startDateLabel && endDateLabel
      ? <div className={css.labels}>
          <label className={startDateLabelClasses} htmlFor={startDateId}>{startDateLabel}</label>
          <label className={endDateLabelClasses} htmlFor={endDateId}>{endDateLabel}</label>
        </div>
      : null;

    // eslint-disable-next-line no-unused-vars
    const { onBlur, onFocus, ...restOfInput } = input;
    const inputProps = {
      onBlur: this.handleBlur,
      onFocus: this.handleFocus,
      ...restOfInput,
      ...rest,
    };
    const classes = classNames(rootClassName || css.fieldRoot, className);

    return (
      <div className={classes}>
        {label}
        <DateRangeInput {...inputProps} />
        <div className={css.inputBorders}>
          <div className={startDateBorderClasses} />
          <div className={endDateBorderClasses} />
        </div>
        <ValidationError fieldMeta={meta} />
      </div>
    );
  }
}

DateRangeInputFieldComponent.defaultProps = {
  rootClassName: null,
  className: null,
  endDateId: null,
  endDateLabel: null,
  endDatePlaceholderText: null,
  startDateId: null,
  startDateLabel: null,
  startDatePlaceholderText: null,
};

const { object, string } = PropTypes;

DateRangeInputFieldComponent.propTypes = {
  rootClassName: string,
  className: string,
  endDateId: string,
  endDateLabel: string,
  endDatePlaceholderText: string,
  startDateId: string,
  startDateLabel: string,
  startDatePlaceholderText: string,
  input: object.isRequired,
  meta: object.isRequired,
};

const DateRangeInputField = props => {
  return <Field component={DateRangeInputFieldComponent} {...props} />;
};

export { DateRangeInput };
export default DateRangeInputField;
