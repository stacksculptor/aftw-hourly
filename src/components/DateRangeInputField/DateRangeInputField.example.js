/* eslint-disable no-console */
import React from 'react';
import { reduxForm, propTypes as formPropTypes } from 'redux-form';
import moment from 'moment';
import { Button } from '../../components';
import { required, bookingDatesRequired } from '../../util/validators';
import DateRangeInputField from './DateRangeInputField';

const FormComponent = props => {
  const { form, handleSubmit, pristine, submitting, dateInputProps } = props;
  const submitDisabled = pristine || submitting;
  return (
    <form onSubmit={handleSubmit}>
      <DateRangeInputField {...dateInputProps} />
      <Button type="submit" disabled={submitDisabled} style={{ marginTop: '24px' }}>
        Select
      </Button>
    </form>
  );
};

FormComponent.propTypes = formPropTypes;

const defaultFormName = 'Styleguide.DateRangeInput.Form';

const Form = reduxForm({
  form: defaultFormName,
})(FormComponent);

export const Empty = {
  component: Form,
  props: {
    dateInputProps: {
      name: 'bookingDates',
      startDateId: `${defaultFormName}.bookingStartDate`,
      startDateLabel: 'Start date',
      startDatePlaceholderText: moment().format('ddd, MMMM D'),
      endDateId: `${defaultFormName}.bookingEndDate`,
      endDateLabel: 'End date',
      endDatePlaceholderText: moment().add(1, 'days').format('ddd, MMMM D'),
      format: null,
      validate: [
        required('Required'),
        bookingDatesRequired('Start date is not valid', 'End date is not valid'),
      ],
      onBlur: () => console.log('onBlur called from DateRangeInput props.'),
      onFocus: () => console.log('onFocus called from DateRangeInput props.'),
    },
    onChange: values => {
      const { startDate, endDate } = values;
      console.log('Changed to', moment(startDate).format('L'), moment(endDate).format('L'));
    },
    onSubmit: values => {
      console.log('Submitting a form with values:', values);
    },
  },
  group: 'custom inputs',
};
