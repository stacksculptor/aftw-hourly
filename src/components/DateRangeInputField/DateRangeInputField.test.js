import React from 'react';
import { renderDeep } from '../../util/test-helpers';
import * as propTypes from '../../util/propTypes';
import { DateRangeInput } from './DateRangeInputField';

const noop = () => null;

describe('DateRangeInput', () => {
  it('matches snapshot', () => {
    const props = {
      unitType: propTypes.LINE_ITEM_NIGHT,
      name: 'bookingDates',
      onBlur: noop,
      onChange: noop,
      onFocus: noop,
      onDragStart: noop,
      onDrop: noop,
      startDateId: 'bookingStartDate',
      startDatePlaceholderText: 'today',
      endDateId: 'bookingEndDate',
      endDatePlaceholderText: 'tomorrow',
    };
    const tree = renderDeep(<DateRangeInput {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
