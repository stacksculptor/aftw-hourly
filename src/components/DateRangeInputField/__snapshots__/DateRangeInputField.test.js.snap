// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeInput matches snapshot 1`] = `
<div
  className=""
>
  <div
    className="DateRangePicker"
  >
    <div>
      <div
        className="DateRangePickerInput"
      >
        <div
          className="DateInput DateInput--open-down"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-bookingStartDate"
            aria-label="today"
            autoComplete="off"
            className="DateInput__input needsclick"
            disabled={false}
            id="bookingStartDate"
            name="bookingStartDate"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="today"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <p
            className="screen-reader-only"
            id="DateInput__screen-reader-message-bookingStartDate"
          >
            DateRangeInputField.screenReaderInputMessage
          </p>
          <div
            className="DateInput__display-text"
          >
            today
          </div>
        </div>
        <div
          aria-hidden="true"
          className="DateRangePickerInput__arrow"
          role="presentation"
        >
          <span />
        </div>
        <div
          className="DateInput DateInput--open-down"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-bookingEndDate"
            aria-label="tomorrow"
            autoComplete="off"
            className="DateInput__input needsclick"
            disabled={false}
            id="bookingEndDate"
            name="bookingEndDate"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="tomorrow"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <p
            className="screen-reader-only"
            id="DateInput__screen-reader-message-bookingEndDate"
          >
            DateRangeInputField.screenReaderInputMessage
          </p>
          <div
            className="DateInput__display-text"
          >
            tomorrow
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
